next.config.mjs

```diff
@@ -1,5 +1,32 @@
 // @ts-check
 
+/*
+ * Netlify UI does not support context-based environment variables.
+ * For example, we cannot configure XYZ=42 for preview releases (pull requests)
+ * and XYZ=4242 for production (main branch).
+ *
+ * As a workaround, we read Netlify’s CONTEXT env variable and then
+ * map PROD_XYZ / XYZ_PROD or PR_XYZ / XYZ_PR into XYZ.
+ */
+
+const suffixOrPrefix =
+  process.env.CONTEXT === "production"
```

Comment:
nit: consider using a hashmap to avoid double ternaries

Reply:
Not sure if this improves a lot in our specific scenario, but yeah, why not? 🙂 Thanks for your feedback!